cranelift-wasm/src/code_translator.rs

```diff
@@ -133,19 +133,23 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
          ***********************************************************************************/
         Operator::Block { ty } => {
             let next = builder.create_ebb();
-            if let Some(ty_cre) = blocktype_to_type(*ty)? {
+            for ty_cre in blocktype_to_types(environ, builder.func, *ty)? {
```

Comment:
This is subtle, but the `next` block here is the basic block we'll jump to for branches out of this wasm `block`. `blocktype_to_types` returns the params of the function type, but what we need here are the returns, because we're creating the block parameters for the block that we'll be exiting to.